model/task_lifecycle.go

```diff
@@ -919,6 +919,13 @@ func UpdateBuildStatus(b *build.Build) (bool, error) {
 		return false, errors.Wrapf(err, "getting tasks in build '%s'", b.Id)
 	}
 
+	b.Aborted = false
+	for _, t := range buildTasks {
+		if t.Aborted {
+			b.Aborted = true
+			break
+		}
+	}
 	buildStatus := getBuildStatus(buildTasks)
 
 	if buildStatus == b.Status {
```

Comment:
If the build status hasn't changed, I think the build aborted status also couldn't have changed, so there's no reason to check for b.Aborted until after this.

Reply:
only updating if status changed